tensorflow/core/kernels/fixed_length_record_reader_op.cc

```diff
@@ -62,14 +64,20 @@ class FixedLengthRecordReader : public ReaderBase {
 
   Status ReadLocked(string* key, string* value, bool* produced,
                     bool* at_end) override {
-    if (input_buffer_->Tell() >= file_pos_limit_) {
+    if (input_buffer_->Tell() >= file_pos_limit_ ||
+        input_buffer_->Tell() + record_bytes_ > file_pos_limit_) {
```

Comment:
When `hop_bytes_ < record_bytes_`, which is true in most cases, the last record will always read some bytes out of `file_pos_limit_` if we only check `input_buffer_->Tell() >= file_pos_limit_`.

For example, assume `hop_bytes_ = 2`, when reading a file with content "0123456FF" ("FF" is "footer"), the last record will be "6FF". If the file is "0123456", an error will be signaled when reading the last record. I thought behaviors like this is not what is expected for a reader with `hop_bytes` argument. It maybe better to have the all bytes of the last record in range of `file_pos_limit_`. What do you think? Thank you for reviewing.